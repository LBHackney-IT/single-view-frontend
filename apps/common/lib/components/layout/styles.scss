@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.mtfh-layout {
  @include govuk-media-query($from: desktop) {
    &__container {
      display: grid;
      grid-row-gap: 0;
      grid-column-gap: 30px;
      grid-template-rows: 1fr;
      grid-template-columns: 310px 1fr;
    }

    &--narrow .mtfh-layout__container {
      grid-template-columns: 1fr;
    }

    &__aside {
      grid-column: 1;
      grid-row: 1;

      .lbh-button {
        min-width: 100%;
      }
    }

    &__main {
      margin-top: 0;
      grid-column: 2;
      grid-row: 1;

      .mtfh-layout.mtfh-layout--narrow & {
        grid-column: 1;
      }
    }

    &--right {
      .mtfh-layout__container {
        grid-template-columns: 1fr 310px;
      }
      .mtfh-layout__main {
        grid-column: 1;
      }
      .mtfh-layout__aside {
        grid-column: 2;
      }
    }
  }
}
