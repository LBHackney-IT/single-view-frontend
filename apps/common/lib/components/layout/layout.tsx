import React, { ComponentPropsWithoutRef, ReactElement, forwardRef } from "react";

import cn from "classnames";

import "./styles.scss";

export interface LayoutProps extends ComponentPropsWithoutRef<"div"> {
  top?: ReactElement;
  backLink?: ReactElement;
  side?: ReactElement;
  sidePosition?: "left" | "right";
}

export const Layout = forwardRef<HTMLDivElement, LayoutProps>(function Layout(
  { children, top, backLink, side, className, sidePosition = "left", ...props },
  ref,
) {
  return (
    <div
      ref={ref}
      className={cn(
        "mtfh-layout",
        { "mtfh-layout--narrow": !side, "mtfh-layout--right": sidePosition === "right" },
        className,
      )}
      {...props}
    >
      {backLink}
      <div id="content" />
      {top}
      <div className="mtfh-layout__container">
        {side ? <div className="mtfh-layout__aside">{side}</div> : null}
        <div className="mtfh-layout__main">{children}</div>
      </div>
    </div>
  );
});
