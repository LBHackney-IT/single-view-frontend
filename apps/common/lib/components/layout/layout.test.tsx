import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Layout } from "./layout";

test("it renders correctly", async () => {
  const { container } = render(
    <Layout
      side={<div id="side">Side Bar</div>}
      top={<div id="top">Top</div>}
      backLink={<div id="backLink">Back Link</div>}
    >
      <div id="main">Main</div>
    </Layout>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders correctly without a side", () => {
  const { container } = render(
    <Layout>
      <div id="main">Main</div>
    </Layout>,
  );
  expect(container).toMatchSnapshot();
});

test("it renders a side on the right", () => {
  const { container } = render(
    <Layout sidePosition="right" side={<div id="side">Side Bar</div>}>
      <div id="main">Main</div>
    </Layout>,
  );
  expect(container).toMatchSnapshot();
});
