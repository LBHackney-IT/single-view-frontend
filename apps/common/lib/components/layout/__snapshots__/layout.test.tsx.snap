// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a side on the right 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--right"
  >
    <div
      id="content"
    />
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__aside"
      >
        <div
          id="side"
        >
          Side Bar
        </div>
      </div>
      <div
        class="mtfh-layout__main"
      >
        <div
          id="main"
        >
          Main
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <div
    class="mtfh-layout"
  >
    <div
      id="backLink"
    >
      Back Link
    </div>
    <div
      id="content"
    />
    <div
      id="top"
    >
      Top
    </div>
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__aside"
      >
        <div
          id="side"
        >
          Side Bar
        </div>
      </div>
      <div
        class="mtfh-layout__main"
      >
        <div
          id="main"
        >
          Main
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders correctly without a side 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--narrow"
  >
    <div
      id="content"
    />
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__main"
      >
        <div
          id="main"
        >
          Main
        </div>
      </div>
    </div>
  </div>
</div>
`;
